### PR TITLE
add match statistic

### PR DESCRIPTION
**Pull request by cgroschupp**

### CODE REVIEW

**Comment by leonnicolas** [comment]
Hey thanks for the contribution.

Do you think we should support the other possible options for the match statistic. I checked out the man page quickly:
```
  statistic
       This module matches packets based on some statistic condition.  It supports two distinct modes settable with the --mode option.

       Supported options:

       --mode mode
              Set the matching mode of the matching rule, supported modes are random and nth.

       [!] --probability p
              Set the probability for a packet to be randomly matched. It only works with the random mode. p must be within 0.0 and 1.0. The supported granularity is in 1/2147483648th increments.

       [!] --every n
              Match one packet every nth packet. It works only with the nth mode (see also the --packet option).

       --packet p
              Set the initial counter value (0 <= p <= n-1, default 0) for the nth mode.
```

I think we should also support `--every` and `--packet` flag.

Or are these flags never returned by `iptables -S` and "translated" to a probability?

**Comment by cgroschupp** [comment]
done, thx for the hint.

File: match_modules.go [hunk 1]
```diff
@@ -39,6 +41,102 @@ func (p *Parser) parseMatch(ms *[]Match) (state, error) {
...
+				}
+			case sINotF:
+				switch {
+				case lit == "--mode":
+					_, lit := p.scanIgnoreWhitespace()
+					(*f)["mode"] = Flag{
+						Not:    true,
+						Values: []string{lit},
+					}
+					s = sStart
```
Comment (leonnicolas) [inline match_modules.go]:
```suggestion
```

I think we can remove this. The `--mode` flag cannot be negated, right?

File: match_modules.go [hunk 2]
```diff
@@ -39,6 +41,102 @@ func (p *Parser) parseMatch(ms *[]Match) (state, error) {
...
+				case lit == "--packet":
+					_, lit := p.scanIgnoreWhitespace()
+					(*f)["packet"] = Flag{
+						Not:    true,
+						Values: []string{lit},
+					}
+					s = sStart
```
Comment (leonnicolas) [inline match_modules.go]:
```suggestion
```

Same thing: `--packet` flag cannot be negated

**Comment by leonnicolas** [comment]
Why did you remove the negated `mode` flag but not the `packat` flag?

**Comment by cgroschupp** [comment]
@leonnicolas sorry, i was beside myself yesterday.

**Comment by leonnicolas** [comment]
Great @cgroschupp Thank you very much!

**Pull request merged**